app/upgrades/v1/bank_send.go

```diff
@@ -0,0 +1,160 @@
+	authkeeper "github.com/cosmos/cosmos-sdk/x/auth/keeper"
+	"github.com/cosmos/cosmos-sdk/x/auth/types"
+	authvesting "github.com/cosmos/cosmos-sdk/x/auth/vesting/types"
+	bankkeeper "github.com/cosmos/cosmos-sdk/x/bank/keeper"
+)
+
+func upgradeBankSend(
+	ctx sdk.Context,
+	authkeeper authkeeper.AccountKeeper,
+	bankkeeper bankkeeper.Keeper,
+	bank_send_list ResultList) error {
+	ctx.Logger().Info(fmt.Sprintf("upgrade :%s", UpgradeName))
+
+	total_allocate_coin := sdk.NewCoin("uguu", sdk.NewInt(0))
+	assumed_coin := sdk.NewCoin("uguu", sdk.NewInt(0))
+
+	// before get total supply
+	before_total_supply := bankkeeper.GetSupply(ctx, "uguu")
+	ctx.Logger().Info(fmt.Sprintf("bank send : total supply[%d]", before_total_supply.Amount))
+
+	// Validator
+	for index, value := range bank_send_list.Validator {
+		ctx.Logger().Info(fmt.Sprintf("bank send validator :%s[%s]", strconv.Itoa(index), value.ToAddress))
+		coin := tokenAllocation(ctx, authkeeper, bankkeeper, index, value)
+		total_allocate_coin.Add(coin)
+	}
+
+	// Check the amount of tokens sent
+	assumed_coin.Add(sdk.NewCoin("uguu", sdk.NewInt(TOTAL_AMOUNT_VALIDATOR)))
+	if !total_allocate_coin.IsEqual(assumed_coin) {
+		panic(fmt.Sprintf("error: assumed amount sent to the validator does not match.: Actual[%d] Assumed[%d]",
+			total_allocate_coin.Amount,
+			assumed_coin.Amount))
+	}
+
+	// Airdrop, Community reward and Moderator
+	for index, value := range bank_send_list.AirdropCommunityRewardModerator {
+		ctx.Logger().Info(fmt.Sprintf("bank send :%s", strconv.Itoa(index)))
+		coin := tokenAllocation(ctx, authkeeper, bankkeeper, index, value)
+		total_allocate_coin.Add(coin)
+	}
+
+	// Check the amount of tokens sent
+	assumed_coin.Add(sdk.NewCoin("uguu", sdk.NewInt(TOTAL_AMOUNT_EXCEPT_VALIDATOR)))
+	if !total_allocate_coin.IsEqual(assumed_coin) {
+		panic(fmt.Sprintf("error: assumed amount sent to the Airdrop, Community reward and Moderator does not match.: Actual[%d] Assumed[%d]",
+			total_allocate_coin.Amount,
+			assumed_coin.Amount))
+	}
+
+	// after get total supply
+	after_total_supply := bankkeeper.GetSupply(ctx, "uguu")
+	ctx.Logger().Info(fmt.Sprintf("bank send : total supply[%d]", after_total_supply.Amount))
+
+	return nil
+}
+
+func tokenAllocation(
+	ctx sdk.Context,
+	authkeeper authkeeper.AccountKeeper,
+	bankkeeper bankkeeper.Keeper,
+	index int,
+	value BankSendTarget) sdk.Coin {
+	not_exist_vesting_account := true
+
+	// check exits VestingAccount
+	fromAddr, err := sdk.AccAddressFromBech32(value.FromAddress)
+	if err != nil {
+		panic(err)
+	}
+	toAddr, err := sdk.AccAddressFromBech32(value.ToAddress)
+	if err != nil {
+		panic(err)
+	}
+	accI := authkeeper.GetAccount(ctx, toAddr)
+	if accI == nil {
+		panic(fmt.Sprintf("error address not exist: [%s][%s]", strconv.Itoa(index), value.ToAddress))
+	}
+
+	cont_acc, ok := accI.(*authvesting.ContinuousVestingAccount)
+	add_coin := sdk.NewCoins(sdk.NewCoin(value.Amount, sdk.NewInt(value.Denom)))
```

Comment:
I think you just need `sdk.Coin` type.
```suggestion
        add_coin := sdk.NewCoin(value.Amount, sdk.NewInt(value.Denom))
```